### PR TITLE
Clickhouse: Seed table when it is empty

### PR DESCRIPTION
**Pull request by sachin9996**
We should be seeding the clickhouse table with data when it is empty, not when no migrations have been applied.

### CODE REVIEW

File: cmd/worker/worker.go [hunk 1]
```diff
@@ -354,6 +360,20 @@ func (c *Command) addClickhouseColumns(ctx context.Context, fields []*tunnel.Eve
 	return nil
 }
 
+func (c *Command) isEventsTableEmpty(ctx context.Context) (bool, error) {
+	var columns uint64
+	if err := c.clickhouse.QueryRow(ctx, `
+		SELECT COUNT(*)
+		FROM INFORMATION_SCHEMA.COLUMNS
+		WHERE (table_name = 'events') AND (table_schema = 'subtrace')
```
Comment (sachin9996) [inline cmd/worker/worker.go:368]: Not sure there's a better way to get the number of columns in a Clickhouse table, I struggled for a bit before I settled on this. `DESCRIBE subtrace.events` doesn't seem to give a result that I can actually count the number of rows. See https://github.com/ClickHouse/ClickHouse/issues/8854 for more details.

File: cmd/worker/worker.go [hunk 2]
```diff
@@ -354,6 +360,20 @@ func (c *Command) addClickhouseColumns(ctx context.Context, fields []*tunnel.Eve
...
+	`).Scan(&columns); err != nil {
+		return false, fmt.Errorf("SELECT COUNT(*): %w", err)
+	}
+	// We expect only the insert_time column to be present when the table is initially created,
+	// see 2024-05-27-events.sql
```
Comment (sachin9996) [inline cmd/worker/worker.go:373]: Not a huge fan of relying on the schema this way but I think it should be fine for now

Comment (sachin9996) [reply cmd/worker/worker.go:373]: Maybe we could filter columns by `/^T([0-9a-fA-F]){8}$/` to see if there are any "event" columns in the table?

File: cmd/worker/clickhouse/clickhouse.go [hunk 3]
```diff
@@ -77,36 +77,34 @@ var migrations embed.FS
 // ApplyMigrations applies all previously unapplied migrations. If successful,
 // it returns the number of migrations previously applied (old) and the number
 // of newly applied migrations.
-func (c *Client) ApplyMigrations(ctx context.Context) (old int, applied int, _ error) {
+func (c *Client) ApplyMigrations(ctx context.Context) (applied int, _ error) {
```
Comment (adtac) [inline cmd/worker/clickhouse/clickhouse.go:80]: Please update the function doc. (Or just remove it, it's kinda obvious with one return variable.)

File: cmd/worker/worker.go (hunk 1 above)
Comment (adtac) [reply cmd/worker/worker.go:368]: `SELECT COUNT(*) FROM system.columns` feels more appropriate. The `information_schema.columns` table seems to have way too many database internals. Plus there's precedent for using `system.columns` in this file; see `hasColumn`.

File: cmd/worker/worker.go (hunk 2 above)
Comment (adtac) [reply cmd/worker/worker.go:373]: Nah, this is fine, we can just change this function if that invariant changes

**Review by adtac — APPROVED** [review]
just a couple of minor nits, feel free to merge and address in a different commit later if that's faster

File: cmd/worker/clickhouse/clickhouse.go (hunk 3 above)
Comment (sachin9996) [reply cmd/worker/clickhouse/clickhouse.go:79]: Changed

**Review by adtac — APPROVED** [review]

**Pull request merged**